activesupport/lib/active_support/core_ext/securerandom.rb

```diff
@@ -22,4 +24,24 @@ def self.base58(n = 16)
       BASE58_ALPHABET[idx]
     end.join
   end
+
+  # SecureRandom.base36 generates a random base36 string in lowercase.
+  #
+  # The argument _n_ specifies the length, of the random string to be generated.
+  #
+  # If _n_ is not specified or is +nil+, 16 is assumed. It may be larger in the future.
+  # This method can be used over +base58+ if a deterministic case key is necessary.
+  #
+  # The result may contain alphanumeric characters in lowercase
+  #
+  #   p SecureRandom.base36 # => "4kugl2pdqmscqtje"
+  #   p SecureRandom.base36(24) # => "77tmhrhjfvfdwodq8w7ev2m7"
+  #
+  def self.base36_lowercase(n = 16)
```

Comment:
Let’s just call this `base36`. It’s enough to document that it returns a lowercase string (as you already have).